tests/e2e/specs/editor-toolbar-buttons.spec.js

```diff
@@ -0,0 +1,246 @@
+import { describe, test, beforeEach } from '@playwright/test';
+import {
+	getCodeMirrorValue,
+	loginToWordPressAdmin,
+	openDrawer,
+	setCodeMirrorValue,
+	typeQuery
+} from "../utils";
+import { expect } from '@playwright/test';
+
+export const selectors = {
+	graphiqlContainer: '.graphiql-container',
+	graphiqlResponse: '.graphiql-response',
+	editorDrawerButton: '.EditorDrawerButton',
+	editorDrawerCloseButton: '.EditorDrawerCloseButton',
+	executeQueryButton: '.graphiql-execute-button',
+	queryInput: '[aria-label="Query Editor"] .CodeMirror',
+	variablesInput: '[aria-label="Variables"] .CodeMirror',
+	prettifyButton: '.graphiql-prettify-button',
+	authButton: '.graphiql-toggle-auth-button',
+};
+
+// Login to WordPress before each test
+beforeEach(async ({ page, context }) => {
+	await loginToWordPressAdmin(page);
+	await context.grantPermissions(['clipboard-read', 'clipboard-write']);
+});
+
+async function openGraphiQL(page) {
+	await expect(page.locator(selectors.graphiqlContainer)).toBeHidden();
+	await openDrawer(page);
+}
+
+describe('Toolbar Buttons', () => {
+
+	beforeEach(async ({ page }) => {
+		await openGraphiQL(page);
+	});
+
+	test('Clicking the Execute button executes a query', async ({ page }) => {
+		await typeQuery(page, 'query { posts { nodes { title } } }');
+		const response = page.locator(selectors.graphiqlResponse);
+		await expect(response).not.toContainText('posts');
+		await page.click(selectors.executeQueryButton);
+		await expect(response).toContainText('posts');
+		await expect(response).toContainText('nodes');
+	});
+
+	describe('Auth button', () => {
+
+		beforeEach(async ({ page }) => {
+			await typeQuery(page, 'query { viewer { name } }');
+		});
+
+		test('Default state is authenticated', async ({ page }) => {
+			// const authButton = page.locator(selectors.authButton);
+
+			// select the 2nd button in the .graphiql-toolbar
+			const authButton = await page.locator( `.graphiql-toolbar button:nth-child(2)` )
```

Comment:
This can likely be added to the `selectors` object at the top of this file.

Using nth-child(2) as a selector feels brittle for something that could be reordered, but we can update this to the generated button class names in https://github.com/wp-graphql/wpgraphql-ide/pull/135 once this is pulled in.